torch/onnx/_internal/fx/onnxfunction_dispatcher.py

```diff
@@ -498,7 +501,7 @@ def perfect_match_inputs(
                 # of this input defined in the OpSchema, we know the function
                 # and the input are not compatible
                 diagnostic.with_additional_message(
-                    f"#### Failed: input type mismatch! \n"
+                    f"#### Failed: input type mismatch for input '{schema_input.name}'! \n"
                     f"Actual {torch_input_compatible_types} vs\n"
```

Comment:
Unrelated question: I recall seeing `torch_input_compatible_types` as empty set when `torch_input` is `None`. Should it count as perfect match if `schema_input` is `optional` or not `required`? @titaiwangms

Reply:
The op doesn't need an input? Could you point me to that op? Do we have optional input cases? I guess I haven't thought of this kind of case.

Reply:
I see some of Optional[INT64]. The logic now relies on the order of the inputs. I will see if I can get the info from OpSchema.

Reply:
<img width="592" alt="image" src="https://github.com/pytorch/pytorch/assets/9376104/9769bbb4-4243-4062-88dc-ad27f48159ef">

Reply:
I think I can add another condition saying that if Optional is allowed, this can be seen as passed. Just wanted to make sure, the optional input still needs to be presenting as "None", right?

Reply:
https://github.com/pytorch/pytorch/issues/106055

Reply:
> I think I can add another condition saying that if Optional is allowed, this can be seen as passed. Just wanted to make sure, the optional input still needs to be presenting as "None", right?

Very likely. I'm unsure about the details, you'd need to check how dispatcher and graph building is handling this today. This op is exported correctly, so probably just need to refine logic in perfect match.

Reply:
Yes. Required Optional[...] inputs needs `None` or `Tensor`